wetterdienst/provider/dwd/observation/api.py

```diff
@@ -515,14 +526,13 @@ def describe_fields(cls, dataset, resolution, period, language: str = "en") -> d
         else:
             raise ValueError("Only language 'en' or 'de' supported")
 
-        file_index = file_index[file_index["filename"].str.contains(file_prefix)]
-
-        description_file_url = str(file_index["filename"].tolist()[0])
+        file_index = file_index.filter(pl.col("filename").str.contains(file_prefix))
+        description_file_url = str(file_index.get_column("filename").item())
         log.info(f"Acquiring field information from {description_file_url}")
 
         return read_description(description_file_url, language=language)
 
-    def _all(self) -> pd.DataFrame:
+    def _all(self) -> pl.LazyFrame:
```

Comment:
Oh, there is a `LazyFrame`?

Reply:
LazyFrame stores all operations added to the frame in a graph and will apply it only after calling `.collect()` . Obviously for a lazyframe you cant access e.g. `.columns` attribute, as in the moment it doesn't know which columns it will have.

Reply:
Isn't there another way to apply lazy computations on column operations as well?

Reply:
What do you mean by that?

Reply:
So, how do I _actually_ display the columns of a `LazyFrame`?

Reply:
I have to correct myself: Actually you can just print the columns like `print(lazy_df.columns)` however it is not possible to set the columns via list like `lazy_df.columns = ["abc", "def"]`. You'd rather have to go like `df.select(pl.col("foo").alias("abc"), pl.col("bar").alias("def"))`